src/main/java/io/swagger/swaggerhub/plugin/requests/SaveSCMPluginConfigRequest.java

```diff
@@ -0,0 +1,248 @@
+package io.swagger.swaggerhub.plugin.requests;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Configuration class used to store details required to create a request to SwaggerHub to create an SCM integration plugin.
+ * The inner builder class follows the builder pattern which aids building the SaveSCMPluginConfigRequest object.
+ * {@link SCMIntegrationPluginConfiguration} represents the object when being sent as part of a request body.
+ */
+public class SaveSCMPluginConfigRequest {
+
+    /*
+    Path Params
+     */
+    private String apiOwner;
+    private String api;
+    private String version;
+    private String oas;
+
+    /*
+    Request body params
+     */
+    private String scmProvider;
+    private String token;
+    private String repository;
+    private String repositoryOwner;
+    private String enabled;
+    private String branch;
+    private String syncMethod;
+    private String outputFile;
+    private String target;
+
+    public SaveSCMPluginConfigRequest(Builder builder) {
+        this.apiOwner = builder.apiOwner;
+        this.api = builder.api;
+        this.version = builder.version;
+        this.oas = builder.oas;
+        this.scmProvider = builder.scmProvider;
+        this.token = builder.token;
+        this.repository = builder.repository;
+        this.repositoryOwner = builder.repositoryOwner;
+        this.enabled = builder.enabled;
+        this.branch = builder.branch;
+        this.syncMethod = builder.syncMethod;
+        this.outputFile = builder.outputFile;
+        this.target = builder.target;
+    }
+
+    public String getApiOwner() {
+        return apiOwner;
+    }
+
+    public String getApi() {
+        return api;
+    }
+
+    public String getVersion() {
+        return version;
+    }
+
+    public String getOas() {
+        return oas;
+    }
+
+    public String getScmProvider() {
+        return scmProvider;
+    }
+
+    public String getToken() {
+        return token;
+    }
+
+    public String getRepository() {
+        return repository;
+    }
+
+    public String getRepositoryOwner() {
+        return repositoryOwner;
+    }
+
+    public String getEnabled() {
+        return enabled;
+    }
+
+    public String getBranch() {
+        return branch;
+    }
+
+    public String getSyncMethod() {
+        return syncMethod;
+    }
+
+    public String getOutputFile() {
+        return outputFile;
+    }
+
+    public String getTarget() {
+        return target;
+    }
+
+    public String getRequestBody() throws JsonProcessingException {
+        return new ObjectMapper().writeValueAsString(new SCMIntegrationPluginConfiguration(this));
+    }
+
+    /**
+     * Data transfer object used to send a potential SCM plugin configuration as a request body on a POST/PUT request
+     */
+    public class SCMIntegrationPluginConfiguration {
```

Comment:
why is this an inner class? I cant see why its not its own class

Reply:
Originally had this as part of a DTO's package but wanted to cut down on the size of the PR.

Reply:
Moving out into dtos package.